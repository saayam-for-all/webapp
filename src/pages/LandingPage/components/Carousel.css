.swiper {
  width: 100%;
  padding-top: 50px;
  padding-bottom: 50px;
}

.swiper-slide {
  background-position: center;
  background-size: cover;
  width: 300px;
  height: 300px;
}

.swiper-slide img {
  display: block;
  width: 100%;
}

.swiper-slide:not(.swiper-slide-active)::after {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 150%;
  background: rgba(0, 0, 0, 0.5); /* adjust darkness */
  border-radius: 16px; /* match your slide border radius */
  z-index: 2;
}

.swiper-slide:not(.swiper-slide-active)::before {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 150%;
  background: rgba(0, 0, 0, 0.5); /* adjust darkness */
  border-radius: 16px; /* match your slide border radius */
  z-index: 2;
}
