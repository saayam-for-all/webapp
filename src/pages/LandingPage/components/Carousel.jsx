import { useEffect, useState } from "react";
import FOOD1 from "../../../assets/carouselImages/food.jpg";
import FOOD2 from "../../../assets/carouselImages/food3.jpg";
import HANDS from "../../../assets/carouselImages/hands.jpg";
import HANDS2 from "../../../assets/carouselImages/hands2.jpg";
import HEALTH from "../../../assets/carouselImages/health.jpg";
import HEALTH2 from "../../../assets/carouselImages/health2.jpg";
import CLOTHES from "../../../assets/carouselImages/clothes.jpg";
import EDUCATION from "../../../assets/carouselImages/education3.jpg";
import REFERRALS from "../../../assets/carouselImages/referrals.jpeg";
import { IoIosArrowBack, IoIosArrowForward } from "react-icons/io";

const CarouselImages = [
  FOOD1,
  FOOD2,
  HANDS,
  HEALTH,
  HANDS2,
  HEALTH2,
  CLOTHES,
  EDUCATION,
  REFERRALS,
];

export default function Carousel() {
  let [current, setCurrent] = useState(0);

  let previousSlide = () => {
    if (current === 0) setCurrent(CarouselImages.length - 1);
    else setCurrent(current - 1);
  };

  let nextSlide = () => {
    if (current === CarouselImages.length - 1) setCurrent(0);
    else setCurrent(current + 1);
  };

  useEffect(() => {
    const intervalId = setInterval(() => {
      nextSlide();
    }, 3000);
    return () => clearInterval(intervalId);
  }, [current]);

  return (
    <div className="overflow-hidden relative h-[500px]">
      <div
        className={`flex transition ease-out duration-1000`}
        style={{
          transform: `translateX(-${current * 100}%)`,
        }}
      >
        {CarouselImages.map((s, idx) => {
          return <img src={s} alt={s} key={idx} />;
        })}
      </div>

      <div className="absolute top-0 h-full w-full justify-between items-center flex text-3xl px-2">
        <div
          onClick={previousSlide}
          className="bg-black opacity-25 hover:opacity-100 rounded-full p-1 text-white hover:bg-slate-700 flex items-center justify-center cursor-pointer"
        >
          <IoIosArrowBack />
        </div>
        <div
          onClick={nextSlide}
          className="bg-black opacity-25 hover:opacity-100 rounded-full p-1 text-white hover:bg-slate-700 flex items-center justify-center cursor-pointer"
        >
          <IoIosArrowForward />
        </div>
      </div>

      <div className="absolute bottom-0 py-4 flex justify-center gap-3 w-full">
        {CarouselImages.map((s, i) => {
          return (
            <div
              onClick={() => {
                setCurrent(i);
              }}
              key={"circle" + i}
              className={`rounded-full w-3 h-3 cursor-pointer  ${
                i == current ? "bg-white" : "bg-gray-500"
              }`}
            ></div>
          );
        })}
      </div>
    </div>
  );
}
