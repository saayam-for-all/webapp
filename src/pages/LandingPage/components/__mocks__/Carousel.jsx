export default function Carousel() {
  return <mock-carousel />;
}
