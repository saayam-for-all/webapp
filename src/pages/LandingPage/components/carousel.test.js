import { render } from "@testing-library/react";
import Carousel from "./Carousel";

describe("Carousel", () => {
  it("renders correctly", () => {
    const tree = render(<Carousel />);
    expect(tree).toMatchSnapshot();
  });
});
