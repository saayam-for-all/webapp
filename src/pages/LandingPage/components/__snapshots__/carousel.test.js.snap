// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="overflow-hidden relative h-[500px]"
      >
        <div
          class="flex transition ease-out duration-1000"
          style="transform: translateX(-0%);"
        >
          <img
            alt="food.jpg"
            src="food.jpg"
          />
          <img
            alt="food3.jpg"
            src="food3.jpg"
          />
          <img
            alt="hands.jpg"
            src="hands.jpg"
          />
          <img
            alt="health.jpg"
            src="health.jpg"
          />
          <img
            alt="hands2.jpg"
            src="hands2.jpg"
          />
          <img
            alt="health2.jpg"
            src="health2.jpg"
          />
          <img
            alt="clothes.jpg"
            src="clothes.jpg"
          />
          <img
            alt="education3.jpg"
            src="education3.jpg"
          />
          <img
            alt="referrals.jpeg"
            src="referrals.jpeg"
          />
        </div>
        <div
          class="absolute top-0 h-full w-full justify-between items-center flex text-3xl px-2"
        >
          <div
            class="bg-black opacity-25 hover:opacity-100 rounded-full p-1 text-white hover:bg-slate-700 flex items-center justify-center cursor-pointer"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
              />
            </svg>
          </div>
          <div
            class="bg-black opacity-25 hover:opacity-100 rounded-full p-1 text-white hover:bg-slate-700 flex items-center justify-center cursor-pointer"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
              />
            </svg>
          </div>
        </div>
        <div
          class="absolute bottom-0 py-4 flex justify-center gap-3 w-full"
        >
          <div
            class="rounded-full w-3 h-3 cursor-pointer  bg-white"
          />
          <div
            class="rounded-full w-3 h-3 cursor-pointer  bg-gray-500"
          />
          <div
            class="rounded-full w-3 h-3 cursor-pointer  bg-gray-500"
          />
          <div
            class="rounded-full w-3 h-3 cursor-pointer  bg-gray-500"
          />
          <div
            class="rounded-full w-3 h-3 cursor-pointer  bg-gray-500"
          />
          <div
            class="rounded-full w-3 h-3 cursor-pointer  bg-gray-500"
          />
          <div
            class="rounded-full w-3 h-3 cursor-pointer  bg-gray-500"
          />
          <div
            class="rounded-full w-3 h-3 cursor-pointer  bg-gray-500"
          />
          <div
            class="rounded-full w-3 h-3 cursor-pointer  bg-gray-500"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="overflow-hidden relative h-[500px]"
    >
      <div
        class="flex transition ease-out duration-1000"
        style="transform: translateX(-0%);"
      >
        <img
          alt="food.jpg"
          src="food.jpg"
        />
        <img
          alt="food3.jpg"
          src="food3.jpg"
        />
        <img
          alt="hands.jpg"
          src="hands.jpg"
        />
        <img
          alt="health.jpg"
          src="health.jpg"
        />
        <img
          alt="hands2.jpg"
          src="hands2.jpg"
        />
        <img
          alt="health2.jpg"
          src="health2.jpg"
        />
        <img
          alt="clothes.jpg"
          src="clothes.jpg"
        />
        <img
          alt="education3.jpg"
          src="education3.jpg"
        />
        <img
          alt="referrals.jpeg"
          src="referrals.jpeg"
        />
      </div>
      <div
        class="absolute top-0 h-full w-full justify-between items-center flex text-3xl px-2"
      >
        <div
          class="bg-black opacity-25 hover:opacity-100 rounded-full p-1 text-white hover:bg-slate-700 flex items-center justify-center cursor-pointer"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
            />
          </svg>
        </div>
        <div
          class="bg-black opacity-25 hover:opacity-100 rounded-full p-1 text-white hover:bg-slate-700 flex items-center justify-center cursor-pointer"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
            />
          </svg>
        </div>
      </div>
      <div
        class="absolute bottom-0 py-4 flex justify-center gap-3 w-full"
      >
        <div
          class="rounded-full w-3 h-3 cursor-pointer  bg-white"
        />
        <div
          class="rounded-full w-3 h-3 cursor-pointer  bg-gray-500"
        />
        <div
          class="rounded-full w-3 h-3 cursor-pointer  bg-gray-500"
        />
        <div
          class="rounded-full w-3 h-3 cursor-pointer  bg-gray-500"
        />
        <div
          class="rounded-full w-3 h-3 cursor-pointer  bg-gray-500"
        />
        <div
          class="rounded-full w-3 h-3 cursor-pointer  bg-gray-500"
        />
        <div
          class="rounded-full w-3 h-3 cursor-pointer  bg-gray-500"
        />
        <div
          class="rounded-full w-3 h-3 cursor-pointer  bg-gray-500"
        />
        <div
          class="rounded-full w-3 h-3 cursor-pointer  bg-gray-500"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
